### PR TITLE
fused_rope supports static graph

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features
### PR changes

APIs
### Description

fused_rope supports static graph
Pcard-76459

### CODE REVIEW

File: test/legacy_test/test_fused_rotary_position_embedding.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def mult_qkv_rotate_half(value, cos_tensor, sin_tensor):
     return query
 
 
-def get_sin_cos_tensor(seq_len, head_dim, sign):
```
Comment (AnnaTrainingG) [inline test/legacy_test/test_fused_rotary_position_embedding.py:62]: 那这个sign 是不是去掉就行了

File: test/legacy_test/test_fused_rotary_position_embedding.py [hunk 2]
```diff
@@ -234,10 +223,14 @@ def test_fused_rope(self):
...
-            paddle_fused_rotary_position_embedding, seed=self.seed
+            paddle_fused_rotary_position_embedding,
+            seed=self.seed,
+            with_sin_cos=True,
         )
         f_fw, f_bw = self.get_forward_backward(
-            fused_rotary_position_embedding, seed=self.seed, flag=True
+            fused_rotary_position_embedding,
+            seed=self.seed,
+            with_sin_cos=True,
```
Comment (niuliling123) [inline test/legacy_test/test_fused_rotary_position_embedding.py:233]: sin cos = None的没验

File: test/legacy_test/test_fused_rotary_position_embedding.py [hunk 3]
```diff
@@ -289,13 +281,59 @@ def test_fused_rope_position_ids(self):
...
+            shape=(1, tensor_q.shape[1], 1, tensor_q.shape[3]),
+            dtype=self.dtype,
+        )
+
+        out_q, out_k, out_v = fused_rotary_position_embedding(
+            q,
+            k,
+            v,
+            sin,
+            cos,
```
Comment (AnnaTrainingG) [inline test/legacy_test/test_fused_rotary_position_embedding.py:315]: sin,cos = None  / use_neox_rotary_style = True的没有

File: test/legacy_test/test_fused_rotary_position_embedding.py (hunk 1 above)
Comment (zhiqiu) [reply test/legacy_test/test_fused_rotary_position_embedding.py:62]: 是的，后面清理下

**Review by AnnaTrainingG — APPROVED** [review]
LGTM

**Pull request merged**